src/main/java/frc/robot/commands/CompositeCommands.java

```diff
@@ -0,0 +1,43 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import frc.robot.Constants;
+import frc.robot.subsystems.intake.Intake;
+import frc.robot.subsystems.notepath.Notepath;
+import frc.robot.subsystems.shooter.ShooterBed;
+import frc.robot.subsystems.shooter.ShooterFlywheel;
+
+public final class CompositeCommands
+{
+    private CompositeCommands()
+    {
+    }
+
+    public static Command intakePickup(Intake intake, Notepath notepath)
+    {
+        return Commands.parallel(IntakeCommands.startIntake(intake), NotepathCommands.intakePickup(notepath)).andThen(Commands.waitUntil(() -> notepath.hasNote())).finallyDo(() ->
+        {
+            intake.setIntakeOff();
+            notepath.setOff();
+        });
+    }
+
+    public static Command stopIntaking(Intake intake, Notepath notepath)
+    {
+        return Commands.parallel(IntakeCommands.stopIntake(intake), NotepathCommands.stopFeed(notepath));
+    }
+
+    public static Command shooterPickup(ShooterBed shooterBed, ShooterFlywheel shooterFlywheel, Notepath notepath)
+    {
+        return Commands.parallel(
+                ShooterBedCommands.setBedAngle(shooterBed, Constants.ShooterBed.BED_PICKUP_ANGLE),
+                ShooterFlywheelCommands.shooterFlywheelShoot(shooterFlywheel, -Constants.ShooterFlywheel.FLYWHEEL_INTAKE_SPEED, -Constants.ShooterFlywheel.FLYWHEEL_INTAKE_SPEED), NotepathCommands.reverseFeed(notepath)
+        ).andThen(Commands.waitUntil(() -> notepath.hasNote())).finallyDo(() ->
```

Comment:
Here's a bit of an odd sequence:

The light sensor that tells us we have a note is located between the notepath and the shooter flywheel. When intaking through the shooter, we can't stop the motors when the light sensor trips because the note will still be hanging halfway out of the shooter flywheel at that point. We actually want to make sure we shut everything off once we no longer see a note.

The tricky part is the "no longer" part. We have to first see a note, then we have to stop seeing it. I think that would look something like this:

```suggestion
        ).andThen(Commands.waitUntil(() -> notepath.hasNote())).andThen(Commands.waitUntil(() -> !notepath.hasNote())).finallyDo(() ->
```

Reply:
Another thing to consider is that we can't necessarily rely on the light sensor to directly tell us we have a note. If we pick up through the intake, we stop when the sensor is tripped. If we pick up through the shooter, we stop when the sensor is NOT tripped. In both cases, we'll have a note, but the sensor state doesn't directly reflect that.

I'm wondering if we need to split apart the logic that says we have a note from the logic that says whether the sensor is tripped or not.